dotCMS/src/main/java/com/dotcms/browser/BrowserAPI.java

```diff
@@ -0,0 +1,583 @@
+            
+            
+            if(null==contentlet) {
+                return;
+            }
+
+            wfActions = APILocator.getWorkflowAPI().findAvailableActions(contentlet, user, WorkflowAPI.RenderMode.LISTING);
+            
+            if (permissionAPI.doesUserHavePermission(contentlet, PermissionAPI.PERMISSION_WRITE, user) && contentlet.isLocked()) {
+                String lockedUserId = APILocator.getVersionableAPI()
+                        .getLockedBy(contentlet);
+                if (user.getUserId().equals(lockedUserId)) {
+                    contentEditable = true;
+                } else {
+                    contentEditable = false;
+                }
+            } else {
+                contentEditable = false;
+            }
+            
+
+
+
+            if (permissions.contains(PERMISSION_READ)) {
+                if (!showArchived && contentlet.isArchived()) {
+                    skip=true;
+                    return;
+                }
+                final boolean showScheme = (wfActions!=null) ?  wfActions.stream().collect(Collectors.groupingBy(WorkflowAction::getSchemeId)).size()>1 : false;
+                
+                for (final WorkflowAction action : wfActions) {
+
+					WorkflowScheme wfScheme = APILocator.getWorkflowAPI().findScheme(action.getSchemeId());
+                    Map<String, Object> wfActionMap = new HashMap<String, Object>();
+                    wfActionMap.put("name", action.getName());
+                    wfActionMap.put("id", action.getId());
+                    wfActionMap.put("icon", action.getIcon());
+                    wfActionMap.put("assignable", action.isAssignable());
+                    wfActionMap.put("commentable", action.isCommentable()
+                            || UtilMethods.isSet(action.getCondition()));
+
+                    final String actionName = Try.of(() ->LanguageUtil.get(user, action.getName())).getOrElse(action.getName());
+                    final String schemeName = Try.of(() ->LanguageUtil.get(user,wfScheme.getName())).getOrElse(wfScheme.getName());
+                    
+                    final String actionNameStr = (showScheme) ? actionName +" ( "+schemeName+" )" : actionName;
+
+                    wfActionMap.put("wfActionNameStr",actionNameStr);
+                    wfActionMap.put("hasPushPublishActionlet", action.hasPushPublishActionlet());
+                    wfActionMapList.add(wfActionMap);
+                }
+
+            }
+
+
+        }
+    }
+    @Deprecated
+    public Map<String, Object> getFolderContent(User user, String folderId,
+                    int offset, int maxResults, String filter, List<String> mimeTypes,
+                    List<String> extensions, boolean showWorking, boolean showArchived, boolean noFolders,
+                    boolean onlyFiles, String sortBy, boolean sortByDesc, boolean excludeLinks, long languageId)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Parameter 'sortByDesc' is not assigned and could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5337824)